const Contact = require("./Contact");

// Get All Contacts=> ðŸ¥ˆ

exports.getAllContact = (req, res) => {
  Contact.find()
    .then((contacts) => {
      // res.json(contacts);
      res.render("index", { contacts, error: {} });
    })
    .catch((error) => {
      console.log(error);
      res.json({
        message: "Error Occurred ðŸ’¥",
      });
    });
};

// Get Single Contact => ðŸ¥‰
exports.getSingleContact = (req, res) => {
  let { id } = req.params;
  Contact.findById(id)
    .then((contact) => {
      res.json(contact);
    })
    .catch((error) => {
      console.log(error);
      res.json({
        message: "Error Occurred ðŸ’¥",
      });
    });
};

// Create Contact => ðŸ¥‡
exports.createContact = (req, res) => {
  let { name, phone, id, email } = req.body;

  let error = {};
  if (!name) {
    error.name = "Please Provide a Name ðŸ’¥";
  }

  if (!phone) {
    error.phone = "Please Provide a Phone Number ðŸ’¥";
  }

  if (!email) {
    error.email = "Please Provide a Email ðŸ’¥";
  }

  let isError = Object.keys(error).length > 0;
  if (isError) {
    Contact.find()
      .then((contacts) => {
        return res.render("index", { contacts, error });
      })
      .catch((error) => {
        console.log(error);
        return res.json({
          message: "Error Occurred ðŸ’¥",
        });
      });
  }

  let contact = new Contact({
    name,
    email,
    phone,
  });
  contact
    .save()
    .then((contact) => {
      Contact.find().then((contacts) => {
        return res.render("index", { contacts, error: {} });
      });
    })
    .catch((error) => {
      console.log(error);
      return res.json({
        message: "Error Occurred ðŸ’¥",
      });
    });
};

// Update Contact => ðŸ¥‰
exports.updateContact = (req, res) => {
  let { name, email, phone } = req.body;
  let { id } = req.params;

  Contact.findOneAndUpdate(
    { _id: id },
    { $set: { name, email, phone } },
    { new: true }
  )
    .then((contact) => {
      res.json(contact);
    })
    .catch((error) => {
      console.log(error);
      res.json({
        message: "Error Occurred ðŸ’¥",
      });
    });
};

// Delete Contact => ðŸ¥‡
exports.deleteContact = (req, res) => {
  let { id } = req.params;
  Contact.findOneAndDelete({ _id: id })
    .then((contact) => {
      res.json(contact);
    })
    .catch((error) => {
      console.log(error);
      res.json({
        message: "Error Occurred ðŸ’¥",
      });
    });
};
